projects/rates.go

```diff
@@ -0,0 +1,1388 @@
+// in the smallest currency unit (e.g., cents). For example, €10.00 is 1000.
+func NewRateProjectUserUpdateRequest(projectID int64, userID int64, rate *int64) RateProjectUserUpdateRequest {
+	return RateProjectUserUpdateRequest{
+		Path: RateProjectUserUpdateRequestPath{
+			ProjectID: projectID,
+			UserID:    userID,
+		},
+		UserRate: rate,
+	}
+}
+
+// HTTPRequest creates an HTTP request for the RateProjectUserUpdateRequest.
+func (r RateProjectUserUpdateRequest) HTTPRequest(ctx context.Context, server string) (*http.Request, error) {
+	uri := fmt.Sprintf("%s/projects/api/v3/rates/projects/%d/users/%d.json", server, r.Path.ProjectID, r.Path.UserID)
+
+	var body bytes.Buffer
+	if err := json.NewEncoder(&body).Encode(r); err != nil {
+		return nil, fmt.Errorf("failed to encode update project user rate request: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPut, uri, &body)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	return req, nil
+}
+
+// RateProjectUserUpdateResponse represents the response for updating a project user rate.
+type RateProjectUserUpdateResponse struct {
+	// UserRate is the user's updated rate as a monetary amount in the smallest
+	// currency unit (e.g., cents).
+	UserRate int64 `json:"userRate"`
+
+	// Rate is the rate as a monetary amount in the smallest currency unit
+	// (e.g., cents).
+	Rate int64 `json:"rate"`
+
+	// Included contains related data.
+	Included struct {
+		Currencies map[string]Currency `json:"currencies"`
+	} `json:"included"`
+}
+
+// HandleHTTPResponse handles the HTTP response for the RateProjectUserUpdateResponse.
+func (r *RateProjectUserUpdateResponse) HandleHTTPResponse(resp *http.Response) error {
+	if resp.StatusCode != http.StatusCreated {
+		return twapi.NewHTTPError(resp, "failed to update project user rate")
+	}
+
+	// API returns 201 with no content, so no need to decode response body
+	return nil
+}
+
+// RateProjectUserUpdate updates a project user rate using the provided request and returns the response.
+func RateProjectUserUpdate(
+	ctx context.Context,
+	engine *twapi.Engine,
+	req RateProjectUserUpdateRequest,
+) (*RateProjectUserUpdateResponse, error) {
+	return twapi.Execute[RateProjectUserUpdateRequest, *RateProjectUserUpdateResponse](ctx, engine, req)
+}
+
+// RateProjectUserHistoryGetRequestPath contains the path parameters for getting project user rate history.
+type RateProjectUserHistoryGetRequestPath struct {
+	// ProjectID is the unique identifier of the project.
+	ProjectID int64
+
+	// UserID is the unique identifier of the user.
+	UserID int64
+}
+
+type RateProjectUserHistoryGetRequestSideload string
+
+const (
+	RateProjectUserHistoryGetRequestSideloadCurrencies RateProjectUserHistoryGetRequestSideload = "currencies"
+	RateProjectUserHistoryGetRequestSideloadUsers      RateProjectUserHistoryGetRequestSideload = "users"
+)
```

Comment:
Missing comments in the exported types. It's important to add those, as they will show up in the Go documentation, and can help users to understand when to use them.